chainercv/utils/iterator/apply_to_batch.py

```diff
@@ -0,0 +1,153 @@
+from chainercv.utils.iterator.unzip import unzip
+
+
+def apply_to_batch(func, iterator, n_input=1, hook=None):
+    """Apply a function/method to an iterator of batches.
+
+    This function applies a function/method to an iterator of batches.
+    It assumes that the iterator returns a batch of input data or
+    a batch of tuples whose first elements ara input data.
+
+    >>> batch = next(iterator)
+    >>> # batch: [in_val]
+    or
+    >>> # batch: [(in_val0, in_val1, ...)]
+    or
+    >>> # batch: [(in_val0, in_val1, ..., rest_val0, rest_val1, ...)]
+
+    This function applies :func:`func` to batch(es) of input data and gets
+    computed value(s). :func:`func` should take a batch of data and
+    return a batch of computed values
+    or a tuple of batches of computed values.
+
+    >>> out_vals = func(in_val0, in_val1, ...)
+    >>> # out_vals: [out_val]
+    or
+    >>> out_vals0, out_vals1, ... = func(in_val0, in_val1)
```

Comment:
`in_val0, in_val1` --> `in_val0, in_val1, ...` for consistency

Reply:
I think these examples are not necessary.

Reply:
I would like to show the returned values should be `([out_val0], [out_val1], ...)` and not `[(out_val0, out_val1, ...)]`. I feel it is unclear without examples.

Reply:
Ah. I see. So you wanted to emphasize that the function returns tuple of lists.

BTW, shouldn't function take list of values?
So ` out_vals0, out_vals1, ... = func(in_val0, in_val1)` --> ` out_vals0, out_vals1, ... = func(in_vals0, in_vals1)`.

Reply:
You are right. That's my mistake.